### ISSUE TITLE
Default naming conventions

### ISSUE THREAD

**Issue by adireddy** [adireddy/haxe-checkstyle#15]
@AlexHaxe Hey Alex, I have just noticed that you are allowing underscore `_` for lot of naming checks like Classes, Typedefs, Interfaces, Parameters, Members, etc.

Can you please change that and not allow underscore by default. Underscore should only be used for constants and everything else should be upper/lower camel case.

I don't think we should even allow customisation of these regular expressions but for now I am fine with it.

For example check the java coding conventions
http://www.oracle.com/technetwork/java/codeconventions-135099.html

I also think there is some level of duplication in naming checks and we should refactor them.

Let me know what you think.

**Comment by AlexHaxe** [comment]
Hi, 

yes, there are a few underscores in those expressions, and I should remove them. I left them in there because I was unsure what the actual default for the library should be (and maybe also because of copy&paste).
The checkstyle library itself sometimes uses underscore for private fields, sometimes it doesn't, so it is not really consistent. It also produces almost 300 checkstyle errors when run with config.json and over 400 without any config.
When writing tests I try to mimick existing code style (obviously my private coding style is a bit different), but in some cases it is not always clear what the correct style should be.

I think the customisation of the format expressions is very important, at least that was why I started writing those name checks, because my projects showed lots of false positives with the old NamingCheck (again my naming style looks a bit different). Refactoring my code in all of these projects was not an option, so I was looking into making the expressions customisable (and I looked into the java version of checkstyle to get inspiration about the check names and parameters they use). 
Of course something like the TypeNameCheck might not be too useful, since Haxe already requires types to start with an uppercase letter. But someone might want to enforce that all interface names start with an "I".
To fully replace NamingCheck we need a LocalVariableName check, also we need to define what to do with getters and setters.

In conclusion I think we should:
a) make the defaults of all checks match the project's coding style
b) make checkstyle pass it's own checks

I can try to adjust the defaults of my naming checks, I think I will create a pull request, so you can comment on the changes and I can get your feedback before we merge.

For passing checkstyle's checks we could create a special branch and make a (long term) group effort. Unfortunately there is no way to get travisCI to create checkstyle graphs (at least as far as I know), because then we could add a live image to the readme. Maybe we could look into that

**Comment by adireddy** [comment]
Let's "make checkstyle pass it's own checks" first :) that's a very good point.

Also let's leave the customisation as it is and try to match defaults similar to Java.

**Comment by adireddy** [comment]
Created `refactor` branch.

**Comment by alebianco** [comment]
i found https://coveralls.io/ to keep reports of the unit tests code coverage but nothing for reports or generic reports. We could make travis push a file into the repo eventually

**Comment by AlexHaxe** [comment]
To make a graph, we would have to keep historic data, but pushing current values back to github would create a loop of endless travis builds :)

However if there was a second repository where we kept historical data (pushed from travis without triggering a new build) as csv, json or xml, then we could try to find a tool to create a png (also pushed to second repo) and our readme could link to the raw image data to show current results. 

We might even find a tool to generate a html presentation of the full results, where you could navigate through packages and classes - although since our goal is to get to zero errors, such a generator would be of little use. 
Maybe Java checkstyle has command line tools that we could use.

Also nice shout out from @andyli during his WWX2015 presentation on CI for Haxe projects.

**Comment by adireddy** [comment]
Yes, good to hear @andyli mentioning checkstyle. I will try and speak to him later about checkstyle integration with Travis.
